sentence.go

```diff
@@ -57,14 +60,28 @@ func parseSentence(raw string) (BaseSentence, error) {
 		return BaseSentence{}, fmt.Errorf(
 			"nmea: sentence checksum mismatch [%s != %s]", checksum, checksumRaw)
 	}
+	talker, typ := parsePrefix(fields[0])
 	return BaseSentence{
-		Type:     fields[0],
+		Talker:   talker,
+		Type:     typ,
 		Fields:   fields[1:],
 		Checksum: checksumRaw,
 		Raw:      raw,
 	}, nil
 }
 
+// parsePrefix takes the first field and splits it into a talker id and data type.
+// If it's a proprietary sentence, then return the entire prefix as the type.
```

Comment:
> If it's a proprietary sentence, then return the entire prefix as the type

It looks like the type does not contain the initial `P`, so the type is not the entire prefix.

Reply:
@adrianmo good catch. This was the original behaviour, but I ended up changing it.